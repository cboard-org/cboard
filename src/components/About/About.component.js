import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Typography from 'material-ui/Typography';
import Paper from 'material-ui/Paper';
import messages from './About.messages';
import FullScreenDialog, { FullScreenDialogContent } from '../FullScreenDialog';

import './About.css';

About.propTypes = {
  open: PropTypes.bool,
  onRequestClose: PropTypes.func
};

function About({ open, onRequestClose }) {
  return (
    <FullScreenDialog
      open={open}
      title={<FormattedMessage {...messages.about} />}
      onRequestClose={onRequestClose}
    >
      <Paper>
        <FullScreenDialogContent>
          <Typography type="body1">
            <FormattedMessage {...messages.intro} />
          </Typography>

          <Typography type="headline">
            <FormattedMessage {...messages.contributors} />
          </Typography>
          <Typography type="body1" headlineMapping={{ body1: 'div' }}>
            <ul>
              <li>
                <a href="https://twitter.com/amberleyjohanna">Amberley Romo</a>
              </li>
              <li>
                <a href="https://twitter.com/_arthurdenner">Arthur Denner</a>
              </li>
              <li>
                <a href="https://twitter.com/hwk73">Arijit Bhattacharya</a>
              </li>
              <li>
                <a href="https://github.com/BrendanFDMoore">Brendan Moore</a>
              </li>
              <li>
                <a href="https://twitter.com/jvuillermet">Jeremy Vuillermet</a>
              </li>
              <li>
                <a href="https://twitter.com/jquintozamora">Jose Quinto</a>
              </li>
              <li>Martin Bedouret</li>
              <li>
                <a href="https://github.com/shayc">Shay Cojocaru</a>
              </li>
            </ul>
          </Typography>
          <Typography type="headline">
            <FormattedMessage {...messages.resources} />
          </Typography>
          <Typography type="body1" headlineMapping={{ body1: 'div' }}>
            <ul>
              <li>
                <a href="https://openassistive.org/awesome-assistivetech/">
                  Awesome Assistivetech
                </a>{' '}
                - A curated list of{' '}
                <span role="img" aria-label="cool">
                  ðŸ˜Ž
                </span>{' '}
                awesome Assistive Technology frameworks and tools to help you
                develop your AT tool/system.
              </li>
            </ul>
          </Typography>
          <Typography type="headline">
            <FormattedMessage {...messages.license} />
          </Typography>
          <Typography type="body1" headlineMapping={{ body1: 'div' }}>
            <ul>
              <li>
                Code -{' '}
                <a href="https://github.com/shayc/cboard/blob/master/LICENSE">
                  GPLv3
                </a>
              </li>
              <li>
                Mulberry Symbols -{' '}
                <a href="https://creativecommons.org/licenses/by-sa/2.0/uk/">
                  CC BY-SA
                </a>
              </li>
            </ul>
          </Typography>
        </FullScreenDialogContent>
      </Paper>
    </FullScreenDialog>
  );
}

export default About;
