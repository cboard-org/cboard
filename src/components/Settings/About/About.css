.logo {
  max-width: 40%;
}

.title {
  padding-top: 16px;
}

.imageContainer {
  display: 'flex';
  flex-direction: row;
  flex-wrap: 'wrap';
  justify-content: 'space-around';
  overflow: 'hidden';
  padding: 12px;
}

.imageItem {
  width: 20%;
  padding: 8px;
}
