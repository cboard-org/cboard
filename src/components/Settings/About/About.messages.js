import { defineMessages } from 'react-intl';

export default defineMessages({
  about: {
    id: 'cboard.components.About.about',
    defaultMessage: 'About'
  },
  intro: {
    id: 'cboard.components.About.intro',
    defaultMessage:
      'Cboard is an augmentative and alternative communication (AAC) web application, allowing people with speech and language impairments to communicate by symbols and text-to-speech.'
  },
  contributors: {
    id: 'cboard.components.About.contributors',
    defaultMessage: 'Contributors'
  },
  license: {
    id: 'cboard.components.About.license',
    defaultMessage: 'License'
  },
  resources: {
    id: 'cboard.components.About.resources',
    defaultMessage: 'Resources'
  }
});
