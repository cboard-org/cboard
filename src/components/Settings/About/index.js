export { default } from './About.component';
