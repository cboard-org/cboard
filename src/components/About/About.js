import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Typography from 'material-ui/Typography';
import Paper from 'material-ui/Paper';
import messages from './About.messages';
import FullScreenDialog, { FullScreenDialogContent } from '../FullScreenDialog';

import './About.css';

About.propTypes = {
  history: PropTypes.object.isRequired,
  onRequestClose: PropTypes.func
};

function About({ history, onRequestClose }) {
  return (
    <FullScreenDialog
      open
      title={<FormattedMessage {...messages.about} />}
      onRequestClose={history.goBack}
    >
      <Paper>
        <FullScreenDialogContent>
          <Typography variant="body1">
            <FormattedMessage {...messages.intro} />
          </Typography>

          <Typography variant="headline">
            <FormattedMessage {...messages.contributors} />
          </Typography>
          <Typography variant="body1" headlineMapping={{ body1: 'div' }}>
            <ul>
              <li>
                <a href="https://twitter.com/amberleyjohanna">Amberley Romo</a>
              </li>
              <li>
                <a href="https://twitter.com/_arthurdenner">Arthur Denner</a>
              </li>
              <li>
                <a href="https://twitter.com/hwk73">Arijit Bhattacharya</a>
              </li>
              <li>
                <a href="https://github.com/BrendanFDMoore">Brendan Moore</a>
              </li>
              <li>
                <a href="https://twitter.com/jvuillermet">Jeremy Vuillermet</a>
              </li>
              <li>
                <a href="https://twitter.com/jquintozamora">Jose Quinto</a>
              </li>
              <li>Martin Bedouret</li>
              <li>
                <a href="https://github.com/shayc">Shay Cojocaru</a>
              </li>
            </ul>
          </Typography>
          <Typography variant="headline">
            <FormattedMessage {...messages.resources} />
          </Typography>
          <Typography variant="body1" headlineMapping={{ body1: 'div' }}>
            <ul>
              <li>
                <a href="https://openassistive.org/awesome-assistivetech/">
                  Awesome Assistivetech
                </a>{' '}
                - A curated list of{' '}
                <span role="img" aria-label="cool">
                  ðŸ˜Ž
                </span>{' '}
                awesome Assistive Technology frameworks and tools to help you
                develop your AT tool/system.
              </li>
            </ul>
          </Typography>
          <Typography variant="headline">
            <FormattedMessage {...messages.license} />
          </Typography>
          <Typography variant="body1" headlineMapping={{ body1: 'div' }}>
            <ul>
              <li>
                Code -{' '}
                <a href="https://github.com/shayc/cboard/blob/master/LICENSE">
                  GPLv3
                </a>
              </li>
              <li>
                Mulberry Symbols -{' '}
                <a href="https://creativecommons.org/licenses/by-sa/2.0/uk/">
                  CC BY-SA
                </a>
              </li>
            </ul>
          </Typography>
        </FullScreenDialogContent>
      </Paper>
    </FullScreenDialog>
  );
}

export default About;
