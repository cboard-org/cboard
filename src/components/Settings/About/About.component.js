import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';
import Paper from '@material-ui/core/Paper';

import messages from './About.messages';
import FullScreenDialog, {
  FullScreenDialogContent
} from '../../UI/FullScreenDialog';

import './About.css';

About.propTypes = {
  history: PropTypes.object.isRequired,
  onClose: PropTypes.func
};

function About({ history, onClose }) {
  return (
    <FullScreenDialog
      open
      title={<FormattedMessage {...messages.about} />}
      onClose={history.goBack}
    >
      <Paper>
        <FullScreenDialogContent>
          <Typography variant="body1">
            <FormattedMessage {...messages.intro} />
          </Typography>

          <Typography variant="h5">
            <FormattedMessage {...messages.contributors} />
          </Typography>
          <Typography variant="body1" component="div">
            <ul>
              <li>
                <Link
                  href="https://twitter.com/amberleyjohanna"
                  target="_blank"
                >
                  Amberley Romo
                </Link>
              </li>
              <li>
                <Link href="https://twitter.com/_arthurdenner" target="_blank">
                  Arthur Denner
                </Link>
              </li>
              <li>
                <Link href="https://twitter.com/hwk73" target="_blank">
                  Arijit Bhattacharya
                </Link>
              </li>
              <li>
                <Link href="https://github.com/BrendanFDMoore" target="_blank">
                  Brendan Moore
                </Link>
              </li>
              <li>
                <Link href="https://twitter.com/jvuillermet" target="_blank">
                  Jeremy Vuillermet
                </Link>
              </li>
              <li>
                <Link href="https://twitter.com/jquintozamora" target="_blank">
                  Jose Quinto
                </Link>
              </li>
              <li>
                <Link href="https://github.com/martinbedouret" target="_blank">
                  Martin Bedouret
                </Link>
              </li>
              <li>
                <Link href="https://github.com/shayc" target="_blank">
                  Shay Cojocaru
                </Link>
              </li>
            </ul>
          </Typography>
          <Typography variant="h5">
            <FormattedMessage {...messages.resources} />
          </Typography>
          <Typography variant="body1" component="div">
            <ul>
              <li>
                <Link
                  href="https://openassistive.org/awesome-assistivetech/"
                  target="_blank"
                >
                  Awesome Assistivetech
                </Link>{' '}
                - A curated list of{' '}
                <span role="img" aria-label="cool">
                  ðŸ˜Ž
                </span>{' '}
                awesome Assistive Technology frameworks and tools to help you
                develop your AT tool/system.
              </li>
            </ul>
          </Typography>
          <Typography variant="h5">
            <FormattedMessage {...messages.license} />
          </Typography>
          <Typography variant="body1" component="div">
            <ul>
              <li>
                Code -{' '}
                <Link
                  href="https://github.com/shayc/cboard/blob/master/LICENSE"
                  target="_blank"
                >
                  GPLv3
                </Link>
              </li>
              <li>
                Mulberry Symbols -{' '}
                <Link
                  href="https://creativecommons.org/licenses/by-sa/2.0/uk/"
                  target="_blank"
                >
                  CC BY-SA
                </Link>
              </li>
            </ul>
          </Typography>
        </FullScreenDialogContent>
      </Paper>
    </FullScreenDialog>
  );
}

export default About;
