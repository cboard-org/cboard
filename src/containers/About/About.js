import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Typography from 'material-ui/Typography';
import messages from './messages';
import FullScreenDialog, {
  FullScreenDialogContent
} from '../../components/FullScreenDialog';

import './About.css';

class About extends Component {
  render() {
    const { open, onCancel } = this.props;
    return (
      <FullScreenDialog
        open={open}
        title={<FormattedMessage {...messages.about} />}
        onCancel={onCancel}
      >
        <FullScreenDialogContent>
          <Typography type="body1">
            <FormattedMessage {...messages.intro} />
          </Typography>

          <Typography type="headline">
            <FormattedMessage {...messages.contributors} />
          </Typography>
          <Typography type="body1" headlineMapping={{ body1: 'div' }}>
            <ul>
              <li>Akshat Gupta</li>
              <li>
                <a href="https://twitter.com/amberleyjohanna">Amberley Romo</a>
              </li>
              <li>
                <a href="https://twitter.com/hwk73">Arijit Bhattacharya</a>
              </li>
              <li>
                <a href="https://github.com/BrendanFDMoore">Brendan Moore</a>
              </li>
              <li>
                <a href="https://twitter.com/jvuillermet">Jeremy Vuillermet</a>
              </li>
              <li>Jos√© Quinto</li>
              <li>Martin Bedouret</li>
              <li>
                <a href="https://github.com/shayc">Shay Cojocaru</a>
              </li>
            </ul>
          </Typography>

          <Typography type="headline">
            <FormattedMessage {...messages.license} />
          </Typography>
          <Typography type="body1" headlineMapping={{ body1: 'div' }}>
            <ul>
              <li>
                Code -{' '}
                <a href="https://github.com/shayc/cboard/blob/master/LICENSE">
                  GPLv3
                </a>
              </li>
              <li>
                Symbols -{' '}
                <a href="https://creativecommons.org/licenses/by-sa/2.0/uk/">
                  CC BY-SA
                </a>
              </li>
            </ul>
          </Typography>
        </FullScreenDialogContent>
      </FullScreenDialog>
    );
  }
}

About.propTypes = {
  open: PropTypes.bool,
  onCancel: PropTypes.func
};

export default About;
