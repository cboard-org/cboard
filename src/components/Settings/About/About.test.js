import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';

import About from './About.component';

describe('About tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<About history={{ goBack: () => {} }} />);
  });
});
