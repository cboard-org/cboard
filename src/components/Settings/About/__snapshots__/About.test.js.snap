// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About tests default renderer 1`] = `
<WithStyles(FullScreenDialog)
  onClose={[Function]}
  open={true}
  title={
    <FormattedMessage
      defaultMessage="About"
      id="cboard.components.About.about"
      values={Object {}}
    />
  }
>
  <WithStyles(Paper)>
    <FullScreenDialogContent>
      <WithStyles(Typography)
        variant="body1"
      >
        <FormattedMessage
          defaultMessage="Cboard is an augmentative and alternative communication (AAC) web application, allowing people with speech and language impairments to communicate by symbols and text-to-speech."
          id="cboard.components.About.intro"
          values={Object {}}
        />
      </WithStyles(Typography)>
      <WithStyles(Typography)
        variant="headline"
      >
        <FormattedMessage
          defaultMessage="Contributors"
          id="cboard.components.About.contributors"
          values={Object {}}
        />
      </WithStyles(Typography)>
      <WithStyles(Typography)
        headlineMapping={
          Object {
            "body1": "div",
          }
        }
        variant="body1"
      >
        <ul>
          <li>
            <a
              href="https://twitter.com/amberleyjohanna"
            >
              Amberley Romo
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/_arthurdenner"
            >
              Arthur Denner
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/hwk73"
            >
              Arijit Bhattacharya
            </a>
          </li>
          <li>
            <a
              href="https://github.com/BrendanFDMoore"
            >
              Brendan Moore
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/jvuillermet"
            >
              Jeremy Vuillermet
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/jquintozamora"
            >
              Jose Quinto
            </a>
          </li>
          <li>
            Martin Bedouret
          </li>
          <li>
            <a
              href="https://github.com/shayc"
            >
              Shay Cojocaru
            </a>
          </li>
        </ul>
      </WithStyles(Typography)>
      <WithStyles(Typography)
        variant="headline"
      >
        <FormattedMessage
          defaultMessage="Resources"
          id="cboard.components.About.resources"
          values={Object {}}
        />
      </WithStyles(Typography)>
      <WithStyles(Typography)
        headlineMapping={
          Object {
            "body1": "div",
          }
        }
        variant="body1"
      >
        <ul>
          <li>
            <a
              href="https://openassistive.org/awesome-assistivetech/"
            >
              Awesome Assistivetech
            </a>
             
            - A curated list of
             
            <span
              aria-label="cool"
              role="img"
            >
              ðŸ˜Ž
            </span>
             
            awesome Assistive Technology frameworks and tools to help you develop your AT tool/system.
          </li>
        </ul>
      </WithStyles(Typography)>
      <WithStyles(Typography)
        variant="headline"
      >
        <FormattedMessage
          defaultMessage="License"
          id="cboard.components.About.license"
          values={Object {}}
        />
      </WithStyles(Typography)>
      <WithStyles(Typography)
        headlineMapping={
          Object {
            "body1": "div",
          }
        }
        variant="body1"
      >
        <ul>
          <li>
            Code -
             
            <a
              href="https://github.com/shayc/cboard/blob/master/LICENSE"
            >
              GPLv3
            </a>
          </li>
          <li>
            Mulberry Symbols -
             
            <a
              href="https://creativecommons.org/licenses/by-sa/2.0/uk/"
            >
              CC BY-SA
            </a>
          </li>
        </ul>
      </WithStyles(Typography)>
    </FullScreenDialogContent>
  </WithStyles(Paper)>
</WithStyles(FullScreenDialog)>
`;
